engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ClassBasedValidatorDescriptor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.constraintvalidation;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.EnumSet;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorFactory;
+import javax.validation.constraintvalidation.SupportedValidationTarget;
+import javax.validation.constraintvalidation.ValidationTarget;
+
+import org.hibernate.validator.internal.util.TypeHelper;
+import org.hibernate.validator.internal.util.logging.Log;
+import org.hibernate.validator.internal.util.logging.LoggerFactory;
+
+/**
+ * Represents an implementation of {@link ConstraintValidator}.
+ *
+ * @author Gunnar Morling
+ */
+class ClassBasedValidatorDescriptor<A extends Annotation> implements ConstraintValidatorDescriptor<A> {
+
+	private static final long serialVersionUID = -8207687559460098548L;
+	private static final Log LOG = LoggerFactory.make();
+
+	private final Class<? extends ConstraintValidator<A, ?>> validatorClass;
+	private final Type validatedType;
+	private final EnumSet<ValidationTarget> validationTargets;
+
+	public ClassBasedValidatorDescriptor(Class<? extends ConstraintValidator<A, ?>> validatorClass) {
+		this.validatorClass = validatorClass;
+		this.validatedType = TypeHelper.extractType( validatorClass );
```

Comment:
Did you see my comment about avoiding exposing a new TypeHelper method and use ClassMate instead? Maybe it does not make sense here and has a risk of not being consistent with the rest but I thought I might as well ask.